src/librustdoc/clean/mod.rs

```diff
@@ -959,6 +979,36 @@ fn handle_variant(cx: &DocContext, def: Def) -> Result<(Def, Option<String>), ()
     Ok((parent_def, Some(format!("{}.v", variant.name))))
 }
 
+const PRIMITIVES: &[&str] = &["u8", "u16", "u32", "u64", "u128", "usize",
+                              "i8", "i16", "i32", "i64", "i128", "isize",
+                              "f32", "f64",
+                              "str", "bool", "char"];
+const PRIMITIVES_DEF: &[Def] = &[Def::PrimTy(hir::PrimTy::TyUint(syntax::ast::UintTy::U8)),
```

Comment:
If these listings are supposed to be related to each other, can you at least make it a `&[(&str, Def)]`? A map of some kind would be best, but since it's a const, the slice-of-tuples will work.